cdap-app-templates/cdap-etl/cdap-data-streams-base/src/test/java/io/cdap/cdap/datastreams/DataStreamsTest.java

```diff
@@ -1611,4 +1614,190 @@ public void testRetryTimeout() throws Exception {
+    DataSetManager<Table> outputManager = null;
+    try {
+      Schema schema = Schema.recordOf(
+          "streaming_event_test",
+          Schema.Field.of("id", Schema.of(Schema.Type.STRING)),
+          Schema.Field.of("name", Schema.of(Schema.Type.STRING))
+      );
+      List<StructuredRecord> input = new ArrayList<>();
+      StructuredRecord samuelRecord = StructuredRecord.builder(schema).set("id", "123").set("name", "samuel").build();
+      StructuredRecord jacksonRecord = StructuredRecord.builder(schema).set("id", "456").set("name", "jackson").build();
+      input.add(samuelRecord);
+      input.add(jacksonRecord);
+
+      Set<StructuredRecord> expected = new HashSet<>();
+      expected.add(samuelRecord);
+      expected.add(jacksonRecord);
+
+      String sourceStageName = "streaming_event_source";
+      DataStreamsConfig etlConfig = DataStreamsConfig.builder()
+          .addStage(new ETLStage(
+              sourceStageName, MockStreamingEventSource.getPlugin(schema, input)))
+          .addStage(new ETLStage("streaming_event_sink", MockSink.getPlugin("${streaming_event_output_macro}")))
+          .addConnection(sourceStageName, "streaming_event_sink")
+          .setBatchInterval("1s")
+          .setStopGracefully(false)
+          .build();
+
+      ApplicationId appId = NamespaceId.DEFAULT.app("StreamingEventApp");
+      AppRequest<DataStreamsConfig> appRequest = new AppRequest<>(APP_ARTIFACT, etlConfig);
+      ApplicationManager appManager = deployApplication(appId, appRequest);
+
+      sparkManager = appManager.getSparkManager(DataStreamsSparkLauncher.NAME);
+      // Timeout retry in 1 min and set the base delay to 60s.
+      String outputName = "streaming_event_output";
+      sparkManager.start(ImmutableMap.of("streaming_event_output_macro", outputName));
+      sparkManager.waitForRun(ProgramRunStatus.RUNNING, 10, TimeUnit.SECONDS);
+
+      String keyFormat = "%s.%s";
+      AppStateStore state = getAppStateStore(appId.getNamespace(), appId.getApplication());
+      String completedKey = String.format(keyFormat, sourceStageName,
+          MockStreamingEventSource.MOCK_STREAMING_EVENT_SOURCE_COMPLETED);
+      String startedKey = String.format(keyFormat, sourceStageName,
+          MockStreamingEventSource.MOCK_STREAMING_EVENT_SOURCE_STARTED);
+      String retryKey = String.format(keyFormat, sourceStageName,
+          MockStreamingEventSource.MOCK_STREAMING_EVENT_SOURCE_RETRIED);
+
+      // Wait for the calls
+      Tasks.waitFor(true,
+          () -> {
+            Optional<byte[]> startedState = state.getState(startedKey);
+            Optional<byte[]> completedState = state.getState(completedKey);
+            Optional<byte[]> retryState = state.getState(retryKey);
+            return startedState.isPresent() && completedState.isPresent()
+                && !retryState.isPresent();
+          },
+          3, TimeUnit.MINUTES);
+      state.deleteSate(completedKey);
```

Comment:
typo: deleteSate -> deleteState

can fix in a separate PR

Reply:
Ack, will create a new PR for this.